index.d.ts

```diff
@@ -1,6 +1,7 @@
 import { IncomingMessage, ServerResponse } from 'http'
+import { Plugin } from 'hapi'
```

Comment:
Wouldn't it fail for users not using hapi? Would it be ok to define your own type Plugin to remove hapi dependency?

Reply:
@batrudinych could you return to the original implementation? Looks like it should be something like this:
```typescript
export interface IHapiPlugin {
    register: (server: any, options: IOptions) => object
    name: string
    once: boolean
}

// ...

export declare const hapiPlugin: IHapiPlugin
```